api/rosetta/validator.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package rosetta
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/go-playground/validator/v10"
+
+	"github.com/optakt/flow-dps/rosetta/identifier"
+)
+
+var (
+	// Special case in the validator library that shouldn't really happen with correct usage.
+	errInvalidValidation = errors.New("invalid validation input")
+
+	errBlockLength   = errors.New(blockLength)
+	errAddressLength = errors.New(addressLength)
+	errTxLength      = errors.New(txLength)
+)
+
+// Field names needed by the validator library, not needed or used by our code at the moment.
```

Comment:
Required arguments by the validation library, not used by us. This is what would be seen if one were not using the structured errors returned by the library, but the ordinary `error` value, that has the error as a simple string, saying something like `validation failed for field "X" on tag "Y"`. "X" is the field name defined here.

Reply:
We already talked about it, but I feel like if there are comments you feel like making about the code, they should be in the code comments instead of in PR comments 😄 

If it's stuff related to the PR itself, it can be in its description, but if it's about code, it belongs in the code IMO :p

Reply:
We'll it is in the code 😄 

It documents why the defines are there and that they're not used, this is a little extra backstory I didn't think was really needed there.

Reply:
There is no such thing as `too much backstory` for code comments, really.